sql/core/src/test/scala/org/apache/spark/sql/streaming/TransformWithStateInitialStateSuite.scala

```diff
@@ -83,6 +98,49 @@ abstract class StatefulProcessorWithInitialStateTestClass[V]
   }
 }
 
+/**
+ * Stateful processor that will take a union dataframe output from state data source reader,
+ * with flattened state data source rows.
```

Comment:
Shall we describe what this class will do with initial state briefly, especially about map state? Mapping the first char to double is a neat approach, but they need to read the code to understand what is happening. 

Is it intended to make transformation (changing the key of the map) before setting to state variable? I agree users can make transformation into this method and aren't restricted to feed the same, but one more thing to track to verify the test.

If we just load them as they are into state variables for new query, I'm OK to not add extra explanation.

Reply:
Not much reasons for transforming the keys of map but want to reuse more existing codes (`StatefulProcessorWithInitialStateTestClass`). I've written a new stateful processor and just do the simplest put as you commented above.